runtime/runtime_test.go

```diff
@@ -175,6 +192,49 @@ func TestRuntimeImport(t *testing.T) {
 	assert.Equal(t, interpreter.NewIntValueFromInt64(42), value.Value)
 }
 
+func TestRuntimeCachedProgram(t *testing.T) {
```

Comment:
Nice! It would be great to extend this test to also import another program and verify that the cache is used for it